src/main/java/org/datadog/jenkins/plugins/datadog/model/BuildData.java

```diff
@@ -149,11 +161,13 @@ public BuildData(Run<?, ?> run, @Nullable TaskListener listener) throws IOExcept
         this.tags = DatadogUtilities.getBuildTags(run, envVars);
 
         this.buildUrl = envVars.get("BUILD_URL");
-        if (buildUrl == null) {
-            BuildSpanAction buildSpanAction = run.getAction(BuildSpanAction.class);
-            if (buildSpanAction != null) {
+
+        BuildSpanAction buildSpanAction = run.getAction(BuildSpanAction.class);
+        if (buildSpanAction != null) {
+            if (buildUrl == null) {
                 buildUrl = buildSpanAction.getBuildUrl();
             }
+            version = buildSpanAction.getAndIncrementVersion();
```

Comment:
```suggestion
            this.version = buildSpanAction.getAndIncrementVersion();
```

I thought it was a local variable